### DISCUSSION TITLE
hx -g fetch corrupts git repo

### DISCUSSION THREAD

**Discussion by Zoybean** [helix-editor/helix#10752] [Troubleshooting]
### Summary

Attempting to compile from source corrupts my local git repository, setting the `origin` remote to `tree-sitter-agda`.
Update: this is apparently also reproducible by calling `hx -g fetch`

### Reproduction Steps

 

**I tried this:**

1. `git remote -v`
1. `hx -g fetch`
1. `git remote -v`

**I expected this to happen:**

grammar fetched, remotes unchanged

**Instead, this happened:**

the grammar fetch fails for agda:
```
Fetching 177 grammars
175 up to date git grammars
1 updated grammars
        groovy now on main
Failure 1/1: agda Git command failed.
Stdout:
Stderr: error: Your local changes to the following files would be overwritten by checkout:
        runtime/themes/xoey.toml
Please commit your changes or stash them before you switch branches.
Aborting

Error: 1 grammars failed to fetch
```

and local helix git repo has a different git remote:
```
> git remote -v
origin  https://github.com/tree-sitter/tree-sitter-agda (fetch)
origin  https://github.com/tree-sitter/tree-sitter-agda (push)
```

### Helix log

no applicable entries

### Platform

Windows 10

### Terminal Emulator

Windows Terminal Version: 1.19.11213.0

### Installation Method

cargo install

### Helix Version

24.03-171-g00e9e5ea

**Comment by Zoybean** [comment]
After deleting the existing agda grammar and source, then rerunning ths `hx -g fetch` command, the above no longer occurs. I suspect that the fetch failure somehow left the repo in an intermediate state that was not cleaned up.

**Comment by the-mikedavis** [comment]
I could see this happening if the tree-sitter-agda's `.git` directory was removed somehow but I'm not sure how that could happen

**Reply by Zoybean** [reply]
I'm not sure how it may have happened, as I have never consciously edited that grammar, but I am concerned that this was the result. Could there be a check of some kind in the `--grammar fetch` code, to check that the nested repo's root is where it is expected?

**Reply by the-mikedavis** [reply]
We could do a sanity check that the `runtime/grammars/sources/<name>/.git` file exists and is a directory
